cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterMain.java

```diff
@@ -115,7 +119,13 @@ static Injector createGuiceInjector(CConfiguration cConf) {
       new DiscoveryRuntimeModule().getDistributedModules(),
       new RouterModules().getDistributedModules(),
       new SecurityModules().getDistributedModules(),
-      new IOModule()
+      new IOModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          bind(RouteStore.class).to(ZKRouteStore.class).in(Scopes.SINGLETON);
```

Comment:
@poornachandra RouterMain is used only in distributed mode I think. There was no change required for SDK/InMemory mode.